apps/cowswap-frontend/src/common/pure/OrderProgressBarV2/index.tsx

```diff
@@ -456,25 +445,29 @@ export const FinishedStep: React.FC<FinishedStepProps> = ({ solverCompetition, o
           <styledEl.ReceivedAmount>
             You received <TokenLogo token={order.outputToken} size={16} />{' '}
             <b>
-              {order.kind === 'sell'
-                ? CurrencyAmount.fromRawAmount(
-                    order.outputToken,
-                    order.apiAdditionalInfo?.executedBuyAmount || order.buyAmount || '0'
-                  ).toSignificant(6)
-                : CurrencyAmount.fromRawAmount(
-                    order.outputToken,
-                    order.apiAdditionalInfo?.executedSellAmount || order.sellAmount || '0'
-                  ).toSignificant(6)}{' '}
-              {order.outputToken.symbol}
+              <TokenAmount
+                amount={CurrencyAmount.fromRawAmount(
+                  order.outputToken,
+                  isSellOrder(order.kind)
+                    ? order.apiAdditionalInfo?.executedBuyAmount || order.buyAmount
+                    : order.apiAdditionalInfo?.executedSellAmount || order.sellAmount
+                )}
+                tokenSymbol={order.outputToken}
+              />
             </b>
           </styledEl.ReceivedAmount>
         )}
         {surplusFiatValue ? (
           <styledEl.ExtraAmount>
-            and got an extra <i>+{surplusFiatValue.toFixed(2)} USDC</i> (~${surplusFiatValue.toFixed(2)})
+            and got an extra{' '}
+            <i>
+              +<TokenAmount amount={surplusAmount} tokenSymbol={surplusAmount?.currency} />
+            </i>{' '}
+            (~${surplusFiatValue.toFixed(2)})
           </styledEl.ExtraAmount>
         ) : null}
         {/*TODO: Add states for when there's no surplus and/or when there's a custom recipient*/}
+        {/*TODO: use surplusData.showSurplus flag to determine when there's relevant surplus */}
```

Comment:
@fairlighteth `showSurplus` flag is what you need to decide what to show in regards to a relevant surplus or not